app/controllers/posts_controller.rb

```diff
@@ -7,10 +7,18 @@ def index
 
   def show
     @post = Post.find(params[:id])
-    @restaurant = Restaurant.find(params[:restaurant_id])
+    restaurant = @post.restaurant_id
+    @restaurant = Restaurant.find(restaurant)
   end
 
   def edit
+    @post = Post.find(params[:id])
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    @post.update(title: params[:post][:title], review: params[:post][:review])
```

Comment:
@post.update returns true or false based on if the post was successfully updated or not. I'd add a conditional here to handle the case where update fails

Reply:
Makes sense, will do.